src/Chapter3.hs

```diff
@@ -374,6 +379,26 @@ after the fight. The battle has the following possible outcomes:
 
 -}
 
+data Task2Knight = Task2Knight {
+  task2knightHealth :: Int,
+  task2knightAttack :: Int,
+  task2knightGold   :: Int
+} deriving (Show)
+
+data Task2Monster = Task2Monster {
+  task2monsterHealth :: Int,
+  task2monsterAttack :: Int,
+  task2monsterGold   :: Int
+} deriving (Show)
+
+fight' :: Task2Knight -> Task2Monster -> Int
+fight' knight monster | knightIsVictory  = task2knightGold knight + task2monsterGold monster
+                     | monsterIsVictory = -1
+                     | otherwise        = task2knightGold knight 
+                    where
+                      knightIsVictory = task2knightAttack knight >= task2monsterHealth monster
+                      monsterIsVictory = task2monsterAttack monster >= task2knightHealth knight
```

Comment:
Great solution ⚔️ 

You usually want the alignment not to depend on the previous lines and function names. SO it is better to always start guards at the new lines, so even if you change names of function and variables the function alignment won't change:

```haskell
fight' knight monster 
    | knightIsVictory  = task2knightGold knight + task2monsterGold monster
    | monsterIsVictory = -1
    | otherwise        = task2knightGold knight 
  where
    knightIsVictory = task2knightAttack knight >= task2monsterHealth monster
    monsterIsVictory = task2monsterAttack monster >= task2knightHealth knight
```

Reply:
thanks!
Indeed, the indentation was misaligned as a result of changing the function name.(fight -> fight')